File: exercise.py
def multiplication_table(integer):
    # using for loop
    for i in range(0, 10):  # start from 0 to 9, you can increase the range though...
        answer = i * integer
        print(f"{integer} * {i} = {answer}")

    return


def multiply_table(integer):
    # using while loop
    i = 0

    while i < 10:  # start from 0 to 9, you can increase the range though...
        answer = i * integer
        print(f"{integer} * {i} = {answer}")

        i += 1  # increment by 1

    return


def string_counter(word):
    num = 0

    for _ in word:  # iterate over all the characters '_' in the given word
        num += 1  # increment the value of num by 1 for every character present in the word

    return num


def even_odd(numbers):  # 'numbers' is a list in Python or array in Java
    even = 0  # initialize initial values of even and odd numbers to 0
    odd = 0

    for i in numbers:  # iterate over integers in list of numbers
        if i % 2 == 0:  # if the current integer is divisible by 2
            even += 1  # then increment the 'even' by 1
        else:
            odd += 1  # otherwise increment 'odd' by 1

    return even, odd  # return the cumulative values of even and odd numbers


def remove_duplicates(numbers):  # 'numbers' is a list in Python or array in Java
    new_list = []  # create an empty 'new_list'

    for i in numbers:  # iterate over integers in list of numbers
        if i not in new_list:  # check if the current value of 'i' is already in the 'new_list'
            new_list.append(i)  # add the value of the integer that's not in the 'new_list' to the 'new_list'

    return new_list  # return the 'new_list'


def is_ascending(numbers):  # 'numbers' is a list in Python or array in Java
    isAscending = True  # initialize a boolean variable to False at start

    for i in range(len(numbers) - 1):  # iterate over numbers in the list minus the last element to avoid exceptions
        if numbers[i] > numbers[i + 1]:  # if the current number in the list is greater than the next number in the list
            isAscending = False  # change the boolean value of 'isAscending' to False

    return isAscending  # otherwise return True which is the original value of 'isAscending'


def is_abecedarian(word):  # 'word' is a string of characters to be gotten from the user
    initialCharacter = word[0]  # initialize the first character to 'initialCharacter'

    for character in word:  # iterate over all the characters in the 'word'
        if character < initialCharacter:  # return False if the current character is less than the 'initialCharacter'
            return False

        # assign the current character to 'initialCharacter' and repeat the process
        # until all the characters are iterated over
        initialCharacter = character

    return True


if __name__ == "__main__":
    # number = int(input("Type a number: "))  # convert entry to integer
    # print(multiplication_table(number))
    # print()
    # print(multiply_table(number))

    # print()

    # string = input("Type a word or sentence: ")
    # print(f"{string} has {string_counter(string)} letters!\n")

    # request integer values from the user
    string = input("Enter your array of integers separated by commas ',': ")
    arr = [int(i) for i in string.split(",")]  # convert user input to list/array of integers

    # get the returned values of even and odd numbers and assign them in order
    # even_number, odd_number = even_odd(arr)
    #
    # print(f"\nEven numbers = {even_number}\nOdd numbers = {odd_number}\n")
    #
    # print(f"Old array = {arr}\nCleaned array = {remove_duplicates(arr)}")

    print(is_ascending(arr))

    # request string values from the user and convert them to lowercase
    string = input("Enter a word: ").lower()
    print(is_abecedarian(string))

File: Exercise.java
package exercises;

import java.util.ArrayList;
import java.util.Scanner;

/**
 *
 * @author MOPHE
 */
public class Exercise {

    public static void main(String[] args) {
        // 'number' and 'string' variable initializations
//        int number;
//        String string;

        Scanner scanner = new Scanner(System.in); // scanner object creation from Scanner class

        //        System.out.println("Type a number: ");
        //        number = scanner.nextInt(); // request integer input from the user
        //
        //        multiplicationTable(number);    // calling 'multiplicationTable' method
        //        System.out.println();   // print an empty line for readability
        //        multiplyTable(number);    // calling 'multiplyTable' method
        //
        //        System.out.println();   // print an empty line for readability
        //
        //        Scanner sc = new Scanner(System.in);   // sc object creation from Scanner class
        //        System.out.println("Type a word or sentence: ");
        //        // request string input from the user and return the entire string/sentence
        //        string = sc.nextLine(); // 'next()' return a single word while 'nextLine()' returns an entire sentence
        //
        //        // calling 'stringCounter' method and assigning the return value to 'numberOfCharacters' variable
        //        int numberOfCharacters = stringCounter(string);
        //        System.out.printf("'%s' has %d letters\n", string, numberOfCharacters); // formmated print statement
        // get the size of the array from the user
        System.out.println("How many numbers do you want in your array?: ");
        // initialize the 'size' of array to the user input, and 'i' to 0
        int size = scanner.nextInt(), i = 0;
        int[] array = new int[size];    // create a new array with the inputted 'size' from user input

        System.out.println("Enter your array of integers: ");   // request values to be added to array from user
        while (i < size) {
            array[i] = scanner.nextInt();   // insert the inputted values to the position of 'i'
            i++;    // increment the value of 'i' until it reaches the complete 'size' of the array
        }

//        evenOdd(array); // call the 'evenOdd' method by inserting the array variable as parameter
        // call the 'removeDuplicates' method by inserting the array variable as parameter
//         int[] cleanedArray = removeDuplicates(array);
//        Object[] cleanedArray = removeMultiples(array);
//        System.out.println("Old array = " + Arrays.toString(array));
//        System.out.println("Cleaned array = " + Arrays.toString(cleanedArray));
        printArray(multiplyArray(array));
    }

    /**
     * Prints the formatted multiplication table using for loop
     */
    private static void multiplicationTable(int integer) {
        int answer;

        for (int i = 0; i < 10; i++) {  // loop through 0 to 9 and perform the operation below
            answer = i * integer;   // multiply the value of i by the integer value and assign the result to answer
            System.out.printf("%d * %d = %d\n", integer, i, answer);    // formmated print statement
        }
    }

    /**
     * Prints the formatted multiplication table using while loop
     */
    private static void multiplyTable(int integer) {
        int answer, i = 0;

        while (i < 10) {    // perform the action below 10 times but starting from 0 to 9
            answer = i * integer;   // multiply the value of i by the integer value and assign the result to answer
            System.out.printf("%d * %d = %d\n", integer, i, answer);    // formmated print statement
            i++;    // increment the value of i and perform the operation again until it reaches 10
        }
    }

    /**
     * Returns the total number of characters in a given work
     */
    private static int stringCounter(String word) {
        int count = 0;  // initialize count to initial 0

        for (int i = 0; i < word.length(); i++) {   // iterate through all the characters in the given word
            count++;    // increment the value of count by 1 for any character present
        }

        return count;   // return the accumulated value of count
    }

    /**
     * checks for even and odd numbers in an array and formats the output
     */
    private static void evenOdd(int[] array) {
        int even = 0, odd = 0;

        for (int i = 0; i < array.length; i++) {
            if (array[i] % 2 == 0) {    // if value at 'i' is divisible by 2, it is even, otherwise odd 
                even++;
            } else {
                odd++;
            }
        }

        System.out.printf("Even numbers = %d\nOdd numbers = %d\n\n", even, odd);
    }

    private static int[] removeDuplicates(int[] array) {
        int size = array.length, count, index = 0;
        int[] newArray = new int[size];

        for (int i = 0; i < size; i++) {    // iterate over elements in the provided array
            count = 0;

            for (int j = i + 1; j < size; j++) {    // iterate over all the elements in the array starting from 'i + 1'
                if (array[i] == array[j]) { // if the value at index 'i' of the two iterated arrays are equal,
                    count++;    // then increment the value of 'count' by 1
                    break;  // exit the loop
                }
            }

            if (count == 0) {
                newArray[index] = array[i]; // if the value of 'count' is equal to 0, assign the value at 'i' to new array
                index++;
            }
        }

        // the code below can be written as 'System.arraycopy(newArray, 0, array, 0, index);'
        // used to copy an array from the specified source array, beginning at the specified position,
        // to the specified position of the destination array
//        System.arraycopy(newArray, 0, array, 0, index); // the same as the for loop below...
        for (int i = 0; i < index; i++) {
            array[i] = newArray[i];
        }

        return newArray;    // return the new cleaned array
    }

    private static Object[] removeMultiples(int[] array) {
        int size = array.length;
        ArrayList<Integer> arrayIntegers = new ArrayList<>();

        for (int i = 0; i < size; i++) {
            if (!arrayIntegers.contains(array[i])) {
                arrayIntegers.add(array[i]);
            }
        }

        return arrayIntegers.toArray();
    }

    private static void printArray(int[] array) {
        for (int i = 0; i < array.length; i++) {
            System.out.printf("%d ", array[i]);
        }
    }

    /**
     * Multiply each element in the array by itself.
     *
     * @param array The input array of integers.
     * @return The array with each element multiplied by itself.
     */
    private static int[] multiplyArray(int[] array) {
        for (int i = 0; i < array.length; i++) {
            array[i] = array[i] * array[i];
        }
        return array;
    }
}
